### PR TITLE
Clean up. More cp. mkdir. More test coverage. Fixes.

### PR DESCRIPTION
**Pull request by androbtech**

### CODE REVIEW

File: src/commands/cp.rs [hunk 1]
```diff
@@ -55,26 +123,277 @@ pub fn cp(args: CommandArgs) -> Result<OutputStream, ShellError> {
...
+                            return Err(ShellError::labeled_error(
+                                e.to_string(),
+                                e.to_string(),
+                                name_span,
+                            ));
+                        }
+                        Ok(o) => o,
+                    };
+
+                    for (ref src, ref dst) in sources.translate(|(src, loc)| {
```
Comment (jonathandturner) [inline src/commands/cp.rs]: Can we do these as separate steps? It'd just be easier to read the code if the translate block didn't look like part of the `for`

File: src/commands/mkdir.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    let env = args.env.lock().unwrap();
+    let path = env.path.to_path_buf();
+    let mut full_path = PathBuf::from(path);
+
+    match &args.nth(0) {
+        Some(Tagged { item: value, .. }) => full_path.push(Path::new(&value.as_string()?)),
+        _ => {}
+    }
+
+    if !args.has("p") {
```
Comment (jonathandturner) [inline src/commands/mkdir.rs]: Let's use the full name of the switch, first. We'll have short-name switches in the future we can use to alias them.

Comment (androbtech) [reply src/commands/mkdir.rs]: I was not sure what to name the long form flag. What name do you suggest?

Comment (androbtech) [reply src/commands/mkdir.rs]: I named it `--create-all`

Comment (androbtech) [reply src/commands/mkdir.rs]: nixed `--create-all` and discarded alternative `--deep` by making it the default as discussed.

**Pull request merged**